sky/exceptions.py

```diff
@@ -291,3 +291,8 @@ class PortDoesNotExistError(Exception):
 class UserRequestRejectedByPolicy(Exception):
     """Raised when a user request is rejected by an admin policy."""
     pass
+
+
+class NoClusterLaunchedError(Exception):
```

Comment:
Why needed?

Reply:
This do the optimization we did for previous ray autoscaler-based implementation, i.e. we should not need to invoke the cluster termination if the provisioner never created any resources, e.g., no VPC is found, etc.

Reply:
Added a docstr for this error.